### PR TITLE
Fix copy structure with self returning empty mesh

### PR DESCRIPTION
**Pull request by user27182**
### Overview
Copying `ImageData` structure from itself returns an empty mesh:
``` python
import pyvista as pv
image = pv.ImageData(dimensions=(10, 10, 5))
image.copy_structure(image)
print(image.dimensions)  # (0, 0, 0)
```
This PR fixes this. Discovered as part of #6616

### CODE REVIEW

File: pyvista/core/dataobject.py [hunk 1]
```diff
@@ -676,7 +676,7 @@ def copy_structure(self, dataset: _vtk.vtkDataSet) -> None:
         >>> target.plot(show_edges=True)
 
         """
-        self.CopyStructure(dataset)
+        self.CopyStructure(dataset) if dataset is not self else None
```
Comment (tkoyama010) [inline pyvista/core/dataobject.py:679]:
```suggestion
        self.CopyStructure(dataset) if dataset is not self
```

**Review by tkoyama010 — CHANGES_REQUESTED** [review]

File: pyvista/core/dataobject.py (hunk 1 above)
Comment (user27182) [reply pyvista/core/dataobject.py:679]: I would also prefer this but it's not valid syntax, `else` is always required unfortunately

Comment (tkoyama010) [reply pyvista/core/dataobject.py:679]: Too bad Python is a bit redundant.

**Review by tkoyama010 — APPROVED** [review]
LGTM

**Pull request merged**